Added documentation for the json key new feature (#843)
returntocorp/semgrep-docs: docs/writing-rules/experiments/extract-mode.md

  - `extract`
  - `dest-language`

Extract mode has two **optional** field:
  - `reduce`

>   - `json`

Reviewer: here it's actually transform:
Thinking about it we should have called it  map, so we have a mapreduce inside extract mode

The `dest-language` key is required in extract mode. The value must be a [language tag](/writing-rules/rule-syntax/#language-extensions-and-tags).

### `json`

> The `json` key is optional in extract mode. This boolean value defaults to "false" and it specifies if the extracted content should be parsed as raw source code or as a JSON array. Setting this key to "true" is useful extracting code from json formats such as Jupyter Notebooks.

Reviewer 2: ```suggestion
The `json` is an optional key in the extract mode. This boolean value defaults to `false` and specifies whether the extracted content is parsed as raw source code or as a JSON array. Setting the `json` key to `true` is useful in extracting code from JSON formats such as Jupyter Notebooks.
```

## Limitations of extract mode

> Although extract mode supports JSON array decoding via the `json` key, it does not support other additional processing for the extracted text, such as unescaping strings, which might be useful for some other applications.

Reviewer 2: ```suggestion
Although extract mode supports JSON array decoding with the `json` key, it does not support other additional processing for the extracted text, such as unescaping strings.
```
